### PR TITLE
Add a description to the default GatewayClass

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Adding a description to the default GatewayClass so that some command line UI tools like `k9s` can display description information to help users quickly identify these default resources.

Fixes: #issue-number

```release-note
Add a description to the default GatewayClass.
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-gateway-api-class.yaml [hunk 1]
```diff
@@ -6,5 +6,6 @@ metadata:
   name: cilium
 spec:
   controllerName: io.cilium/gateway-controller
+  description: The default cilium GatewayClass
```
Comment (sayboras) [inline install/kubernetes/cilium/templates/cilium-gateway-api-class.yaml]:
```suggestion
  description: The default Cilium GatewayClass
```

**Review by sayboras — APPROVED** [review]
LGTM, one minor suggestion as per below.

File: install/kubernetes/cilium/templates/cilium-gateway-api-class.yaml (hunk 1 above)
Comment (chaunceyjiang) [reply install/kubernetes/cilium/templates/cilium-gateway-api-class.yaml]: Updated.

**Pull request merged**